### PR TITLE
images: fix Archs expansion in makefile

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:

When we start the new build for `kube-cross` image we got this error

```
docker manifest create --amend gcr.io/k8s-staging-build-image/kube-cross:v1.15.7-canary-1 gcr.io/k8s-staging-build-image/kube-cross-amd64:v1.15.7-canary-1
Created manifest list gcr.io/k8s-staging-build-image/kube-cross:v1.15.7-canary-1
manifest for image gcr.io/k8s-staging-build-image/kube-cross-arm64:v1.15.7-canary-1 does not exist in gcr.io/k8s-staging-build-image/kube-cross:v1.15.7-canary-1
manifest for image gcr.io/k8s-staging-build-image/kube-cross-ppc64le:v1.15.7-canary-1 does not exist in gcr.io/k8s-staging-build-image/kube-cross:v1.15.7-canary-1
make: *** [/workspace/images/build/cross/../../Makefile.common-image:60: manifest] Error 1
ERROR
```

which looks like when we run the `docker manifest create --amend` we need to add all the images for the archs we built.

This piece of the make file https://github.com/kubernetes/release/blob/master/images/Makefile.common-image#L59 add the images to the command, but now we have more `platforms` and the command `subst` just replace one entry and not all entries in the `PLATFORMS` variable

for kube-cross we used to have just one platform and that works fine, but now we introduce more (can see in this PR: https://github.com/kubernetes/release/pull/1853/files) and the command broke.

this PR fix this issue by parsing the `PLATFORMS` and removing the `linux/` from the `PLATFORMS` variable.

before:
```
docker manifest create --amend gcr.io/k8s-staging-build-image/kube-cross:v1.15.7-canary-1 gcr.io/k8s-staging-build-image/kube-cross-amd64:v1.15.7-canary-1
```

after the fix:

```
docker manifest create --amend gcr.io/k8s-staging-build-image/kube-cross:v1.15.7-canary-1 gcr.io/k8s-staging-build-image/kube-cross-amd64:v1.15.7-canary-1 gcr.io/k8s-staging-build-image/kube-cross-arm64:v1.15.7-canary-1 gcr.io/k8s-staging-build-image/kube-cross-ppc64le:v1.15.7-canary-1
```

/assign @justaugustus @saschagrunert @hasheddan @ameukam 

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
images: fix Archs expansion in makefile
```

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
This looks reasonable to me, thanks @cpanato 👍 

/lgtm

**Comment by cpanato** [comment]
will lift the hold since got a review from a sig-release TL 🚀 

/hold cancel

will observe the kube-cross job

**Pull request merged**
